dftimewolf/lib/collectors/grr_hunt.py

```diff
@@ -25,20 +25,63 @@ class GRRHunt(grr_base.GRRBaseModule, module.BaseModule):  # pylint: disable=abs
   """This class groups functions generic to all GRR Hunt modules.
 
   Should be extended by the modules that interact with GRR hunts.
+
+  Attributes:
+    match_mode (str): match mode of the client rule set (ALL or ANY).
+    client_operating_systems (List[str]): a list of client OS types
+        (win, osx or linux).
+    client_labels (List[str]): a list of client labels.
   """
 
   def __init__(self,
                state: DFTimewolfState,
-               name: Optional[str]=None,
-               critical: bool=False) -> None:
+               name: Optional[str],
+               critical: bool = False):
     module.BaseModule.__init__(self, state, name=name, critical=critical)
     grr_base.GRRBaseModule.__init__(self)
+    self.match_mode = ""
+    self.client_operating_systems : List[str] = []
+    self.client_labels : List[str] = []
+
+  def HuntSetup(
+      self,
+      match_mode: Optional[str],
+      client_operating_systems: Optional[str],
+      client_labels: Optional[str]) -> None:
+    """Setup hunt client filter arguments.
+
+    Args:
+      match_mode (Optional[str]): match mode of the client rule set (ALL or
+          ANY).
+      client_operating_systems (Optional[str]): a comma separated list of client
+          OS types (win, osx or linux).
+      client_labels (Optional[str]): a comma separated list of client labels.
+    """
+    if match_mode:
+      if match_mode.upper() not in ('ALL', 'ANY'):
```

Comment:
Super nit: for consistency i'd use either lower or upper throughout this function, but stick to one.

Reply:
Switched to lower for consistency.